### PR TITLE
Annotations (Backend)

### PR DESCRIPTION
**Pull request by Morgazipa**
## About

* **Related Issues:** #181 

Implemented the backend for web annotations, also a added a simple test. The annotation model is described in #181.

### Changes

- Created annotation model
- Implemented custom serializer for W3C compatibility
- Added an endpoint 
- Added a simple test

### CODE REVIEW

File: api/interestr/api/models.py [hunk 1]
```diff
@@ -133,5 +133,19 @@ class DataTemplate(BaseModel):
     def __str__(self):
         return self.name
 
+class AnnoTypes(ChoiceEnum):
+    text = "Text"
+    image = "Image"
+
+class Annotation(BaseModel):
+    anno_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(auth_models.User, related_name='%(app_label)s_%(class)s_related', on_delete=models.SET_NULL, default=None,
```
Comment (ThoAppelsin) [inline api/interestr/api/models.py]: What is `'%(app_label)s_%(class)s'` here?

File: api/interestr/api/serializers.py [hunk 2]
```diff
@@ -233,3 +233,47 @@ class Meta:
...
+            "source":  str(instance.target),
+                "type": str(instance.annotype),
+             "selector":  str(instance.selector)           }
+
+            }
+        return json_str
+
+    def to_internal_value(self, data):
+        
+        profile_id = data["creator"]["id"].strip("http://interestr.com/profile/")
```
Comment (ThoAppelsin) [inline api/interestr/api/serializers.py]: Our website is not at `interestr.com`, yet. For now, we may just strip anything before the `/profile/`.

File: api/interestr/api/views.py [hunk 3]
```diff
@@ -195,6 +195,18 @@ class ProfilePageList(generics.ListAPIView):
     queryset = core_models.ProfilePage.objects.all()
     serializer_class = core_serializers.ProfilePageSerializer
 
+class AnnotationList(generics.ListCreateAPIView):
+    """
+    get:
+    Return a list of all the existing annotations.
+
+    post:
+    Create a new annotatiob instance.
```
Comment (ThoAppelsin) [inline api/interestr/api/views.py:204]: Small typo.

**Review by ThoAppelsin — CHANGES_REQUESTED** [review]

**Comment by ThoAppelsin** [comment]
I have made a couple of remarks, one could possibly be problematic, since we do not really have our website located at `interestr.com`.

Aside from that, I may fail to provide a user id altogether. Could you please tolerate me failing at providing anything at all for the `"creator"` field? You already have made it a nullable field, so please allow me to define it as such, too.

The problem is that I cannot seem to be able to access the user id from the extension. To do that, I may require users to log in from the extension to their accounts, but that will be a lot of work.

**Comment by Morgazipa** [comment]
Good points. I changed the serializer to allow guest annotations.

File: api/interestr/api/models.py (hunk 1 above)
Comment (Morgazipa) [reply api/interestr/api/models.py]: Sorry. Before discussing with you, I created this class as an abstract class which was meant to be inherited from by text and image annotation classes. Thus, different lookup names were required for subclasses.

**Review by ThoAppelsin — APPROVED** [review]

**Pull request merged**